src/ssh-config.ts

```diff
@@ -119,14 +118,14 @@ class SSHConfig extends Array<Line> {
 
     for (const line of this) {
       if (line.type !== LineType.DIRECTIVE) continue
-      if (line.param === 'Host' && glob(line.value, params.Host)) {
+      if (line.param === 'Host' && glob(line.value, host)) {
         setProperty(line.param, line.value)
         for (const subline of (line as Section).config) {
           if (subline.type === LineType.DIRECTIVE) {
             setProperty(subline.param, subline.value)
           }
         }
-      } else if (line.param === 'Match' && 'criteria' in line && match(line.criteria, params)) {
+      } else if (line.param === 'Match' && 'criteria' in line && match(line.criteria, { Host: host })) {
```

Comment:
This was intended to support Match queries such as `.compute({ Host, User })`, I'll add a test case about it.

Reply:
added in master

Reply:
Ahh, gotcha. I went back and forth quite a bit on what type `params` should be. It seemed like it was only ever used as a string to pass as the host. That makes much more sense, I will update it. Should it perhaps also accept `Port`, to simulate what would happen if a user ran something like `ssh ssh://matt@site:2200`?

Reply:
Scratch that, `Match` only has

> **canonical**, **final**, **exec**, **host**, **originalhost**, **user**, and **localuser**

as criteria. Adding a `port` wouldn't make sense.